pkg/kapp/diff/resource_with_history.go

```diff
@@ -80,6 +80,19 @@ func (r ResourceWithHistory) RecordLastAppliedResource(appliedChange Change) (ct
 
 	diff := appliedChange.OpsDiff()
 
+	const resourceSizeMaxLen = 262144
+
+	// kapp deploy should work without adding disable annotation when resource size exceed
+	// (https://github.com/vmware-tanzu/carvel-kapp/issues/410)
+	if len(appliedResBytes) > resourceSizeMaxLen || len(diff.MinimalMD5()) > resourceSizeMaxLen {
+		if resourceWithHistoryDebug {
+			fmt.Printf("Unable to record resource, exceed limit of %d bytes "+
+				"(this might result in the diff showing deletion of fields applied by external agents) \n",
+				resourceSizeMaxLen)
+		}
+		return nil, fmt.Errorf("unable to record resource, limit exceed")
```

Comment:
It would be better if we can mention resource name.

Reply:
This is just for our internal purpose this is not something that will be shown to the user.